### PR TITLE
Add label encoder component to EvalML

### LINKED ISSUE #2852: Add label encoder component to EvalML
Breaking down #2648 to two different issues: 
- one which adds a new LabelEncoder component
- one which uses the label encoder in our pipelines in lieu of using a label encoder attribute

This issue simply tracks the former. :)

### PR DESCRIPTION
**Pull request by angela97lin**
Closes #2852

### CODE REVIEW

**Review by freddyaboulton — APPROVED** [review]
Looks good to me @angela97lin !

File: evalml/pipelines/components/transformers/encoders/label_encoder.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+        """
+        if y is None:
+            raise ValueError("y cannot be None!")
+
+        y_ww = infer_feature_types(y)
+        y_t = self._component_obj.transform(y_ww)
+        y_t = pd.Series(y_t, index=y_ww.index)
+        return X, ww.init_series(y_t)
+
+    def fit_transform(self, X, y):
```
Comment (eccabay) [inline evalml/pipelines/components/transformers/encoders/label_encoder.py:75]: Is there a reason we don't just use the parent `Transformer`'s implementation for this?

**Review by eccabay — APPROVED** [review]
Nice, LGTM!

File: evalml/pipelines/components/transformers/encoders/label_encoder.py (hunk 1 above)
Comment (angela97lin) [reply evalml/pipelines/components/transformers/encoders/label_encoder.py:75]: I think nope, except we want a different docstring 😅 (which was incorrect, updating!)
